### ISSUE TITLE
Godot sporadically freezes when saving.

### ISSUE THREAD

**Issue by leiget** [godotengine/godot#12759]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Xubuntu 16.04.3 LTS
Godot Git build 5fb359d
NVidia gForce 960 GTX

**Issue description:**
When I save or try to run my project, it freezes on the progress bar saying "Saving Scene" and it stays at 100%. It's happens randomly.
Here's the terminal output when it happens:

```
Running: /home/james/Programs/godot_3/godot.x11.tools.64 --path /media/james/F/Godot_James/FPS_Base --remote-debug 127.0.0.1:6007 --allow_focus_steal_pid 24912 --position 1728,240
OpenGL ES 3.0 Renderer: GeForce GTX 960/PCIe/SSE2
GLES3: max ubo light: 409
GLES3: max ubo reflections: 455, ubo size: 144
ARVR: Registered interface: Native mobile
ERROR: ~List: Condition ' _first != __null ' is true.
   At: core/self_list.h:99.
ERROR: getornull: Condition ' !id_map.has(p_rid.get_data()) ' is true. returned: __null
   At: core/rid.h:164.
ERROR: instance_remove_dependency: Condition ' !inst ' is true.
   At: drivers/gles3/rasterizer_storage_gles3.cpp:5826.
```
I've been having random rasterizer errors/warnings when running my projects, so it _may_ have something to do with that, but I have no idea.

Also, in the above terminal output, it says "--position 1728,240" most likely because I have a two monitor setup, a left one at 1280x1024 and a right one at 1920x1080.

**Steps to reproduce:**
Difficult to reproduce, but happens somewhat regularly for me. I have to stop the process in the OS, but it still hangs around in memory. I have to kill it in the task manager to get it to go away completely.

**Link to minimal example project:**
Not sure if it makes a difference, as it happens with just about any project, but here's one I'm working on now that it happens on:
[FPS_Base.zip](https://github.com/godotengine/godot/files/1455963/FPS_Base.zip)

**Comment by eon-s** [comment]
Got this since bullet merge on Ubuntu+Gnome while playing around with kinematic body 3d demo (and lot of body errors but do not seem related).

I was unable to reproduce at will.
@leiget can you try to work a bit on your project using godot physics instead of bullet?

**Comment by leiget** [comment]
@eon-s No, I can push through the bugs, so I'll stick with Bullet for now.

I also noticed that when I run my character into walls, it shutters the whole character. I think it's because the character thinks that a wall is a floor and counts it sometimes. This may be related, or not. Here's a project zip to explain it:
[vibrating_char.zip](https://github.com/godotengine/godot/files/1456347/vibrating_char.zip)

**Comment by Zireael07** [comment]
The character stuttering is probably a duplicate of #11520

**Comment by reduz** [comment]
Does this problem persist? I fixed a few bugs that caused this behavior

**Comment by leiget** [comment]
So far using Godot Physics it's been stable, but I've been building assets for a project in Blender and having been using Godot as much lately. That said, I'm confident the problem is solved as when it was a problem, it would happen the 2nd or 3rd, or even first, time I tried to save. I've tested it with the official beta build and the latest git build, saving many times in a row with both Godot and Bullet selected as the 3D physics engine to use in the project settings window (just in case it mattered), and it never locked up.
